ffi/capi/src/neo_datetime.rs

```diff
@@ -1012,33 +1606,180 @@ pub mod ffi {
             let _infallible = self.0.format(&value).write_to(write);
         }
     }
-}
 
-impl From<ffi::TimePrecision> for icu_datetime::options::TimePrecision {
-    fn from(time_precision: ffi::TimePrecision) -> Self {
-        use icu_datetime::options::TimePrecision;
-        match time_precision {
-            ffi::TimePrecision::Hour => TimePrecision::Hour,
-            ffi::TimePrecision::Minute => TimePrecision::Minute,
-            ffi::TimePrecision::MinuteOptional => TimePrecision::MinuteOptional,
-            ffi::TimePrecision::Second => TimePrecision::Second,
-            ffi::TimePrecision::Subsecond1 => TimePrecision::Subsecond(SubsecondDigits::S1),
-            ffi::TimePrecision::Subsecond2 => TimePrecision::Subsecond(SubsecondDigits::S2),
-            ffi::TimePrecision::Subsecond3 => TimePrecision::Subsecond(SubsecondDigits::S3),
-            ffi::TimePrecision::Subsecond4 => TimePrecision::Subsecond(SubsecondDigits::S4),
-            ffi::TimePrecision::Subsecond5 => TimePrecision::Subsecond(SubsecondDigits::S5),
-            ffi::TimePrecision::Subsecond6 => TimePrecision::Subsecond(SubsecondDigits::S6),
-            ffi::TimePrecision::Subsecond7 => TimePrecision::Subsecond(SubsecondDigits::S7),
-            ffi::TimePrecision::Subsecond8 => TimePrecision::Subsecond(SubsecondDigits::S8),
-            ffi::TimePrecision::Subsecond9 => TimePrecision::Subsecond(SubsecondDigits::S9),
+    #[diplomat::opaque]
+    #[diplomat::rust_link(icu::datetime::FixedCalendarDateTimeFormatter, Struct)]
+    #[diplomat::rust_link(
+        icu::datetime::fieldsets::enums::DateAndTimeFieldSet::zone,
+        FnInStruct,
+        hidden
+    )]
+    #[diplomat::rust_link(icu::datetime::fieldsets::DT::zone, FnInStruct, hidden)]
+    #[diplomat::rust_link(icu::datetime::fieldsets::MDT::zone, FnInStruct, hidden)]
+    #[diplomat::rust_link(icu::datetime::fieldsets::YMDT::zone, FnInStruct, hidden)]
+    #[diplomat::rust_link(icu::datetime::fieldsets::DET::zone, FnInStruct, hidden)]
+    #[diplomat::rust_link(icu::datetime::fieldsets::MDET::zone, FnInStruct, hidden)]
+    #[diplomat::rust_link(icu::datetime::fieldsets::YMDET::zone, FnInStruct, hidden)]
+    #[diplomat::rust_link(icu::datetime::fieldsets::ET::zone, FnInStruct, hidden)]
+    #[diplomat::attr(demo_gen, disable)] // constructors are on a different type :(
+    pub struct NeoZonedDateTimeFormatterGregorian(
+        pub  icu_datetime::FixedCalendarDateTimeFormatter<
+            Gregorian,
+            icu_datetime::fieldsets::enums::CompositeFieldSet,
+        >,
+    );
+
+    impl NeoZonedDateTimeFormatterGregorian {
```

Comment:
thought: I still get confused with the Gregorian at the end. I think we made the right choice, but I still have some residual memory here that's always confusing me :smile: 

(no change requested, just flagging I might mix these up at times)

Reply:
1.x FFI had it at the beginning. I think in 2.x we moved it to the end because

1. They sort together better
2. In Rust the calendar is at the end, since it is a type parameter
3. Keeps the ontology of the type name "more general to more specific"\*

\* although to be most pure, it would probably be `FormatterDateTimeGregorian` (formatter first, then one for datetime, then one for gregorian)